symphony/lib/toolkit/class.field.php

```diff
@@ -990,23 +990,20 @@ public function checkFields(array &$errors, $checkForDuplicates = true)
         $label = $this->get('label');
         $element_name = $this->get('element_name');
 
-        if (Lang::isUnicodeCompiled()) {
-            $valid_name = preg_match('/^[\p{L}]([0-9\p{L}\.\-\_]+)?$/u', $element_name);
-        } else {
-            $valid_name = preg_match('/^[A-z]([\w\d-_\.]+)?$/i', $element_name);
-        }
-
         if ($label === '') {
             $errors['label'] = __('This is a required field.');
         } elseif (strtolower($label) === 'id') {
             $errors['label'] = __('%s is a reserved name used by the system and is not allowed for a field handle. Try using %s instead.', array('<code>ID</code>', '<code>UID</code>'));
+        // Check label starts with a letter
+        } elseif (!preg_match('/^\p{L}/u', $label)) {
+            $errors['label'] = __('The label of the field must begin with a letter.');
         }
 
         if ($element_name === '') {
             $errors['element_name'] = __('This is a required field.');
         } elseif ($element_name === 'id') {
             $errors['element_name'] = __('%s is a reserved name used by the system and is not allowed for a field handle. Try using %s instead.', array('<code>id</code>', '<code>uid</code>'));
-        } elseif (!$valid_name) {
+        } elseif (!preg_match('/^[a-z]/i', $element_name)) {
```

Comment:
I am not sure if this is actually needed. I think that `/^\p{L}/u` would suffice for valid XML. It may be a matter of taste or "best practice", however.

Reply:
I was not comfortable with removing it. Like I've said I am tempted to add to check everywhere the handle is user defined.